src/core/ext/client_channel/client_channel_factory.c

```diff
@@ -85,3 +90,120 @@ grpc_arg grpc_client_channel_factory_create_channel_arg(
   arg.value.pointer.vtable = &factory_arg_vtable;
   return arg;
 }
+
+struct grpc_channel_targets_info {
```

Comment:
Ideally, yes. This implementation uses this naive approach for simplicity, as initially we have only 1-1 name-target mappings.

On a side note, the mdstr_hash_table doesn't look generic enough, due to its usage of mdstr. It's mostly a naming issue, though. 

Let me finish merging and getting it to green and then based on priorities explore changing it. We may find a way to "fix" the mdstr_hash_table.